pkg/asset/installconfig/gcp/client.go

```diff
@@ -243,41 +245,171 @@ func (c *Client) GetPublicDomains(ctx context.Context, project string) ([]string
 	return publicZones, nil
 }
 
+func getDNSZoneByName(ctx context.Context, svc *dns.Service, project, zoneName string) (*dns.ManagedZone, error) {
+	returnedZone, err := svc.ManagedZones.Get(project, zoneName).Context(ctx).Do()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get DNS Zones: %w", err)
+	}
+	return returnedZone, nil
+}
+
 // GetDNSZoneByName returns a DNS zone matching the `zoneName` if the DNS zone exists
 // and can be seen (correct permissions for a private zone) in the project.
 func (c *Client) GetDNSZoneByName(ctx context.Context, project, zoneName string) (*dns.ManagedZone, error) {
-	ctx, cancel := context.WithTimeout(ctx, defaultTimeout)
-	defer cancel()
-
 	svc, err := c.getDNSService(ctx)
 	if err != nil {
 		return nil, err
 	}
-	returnedZone, err := svc.ManagedZones.Get(project, zoneName).Context(ctx).Do()
+	return getDNSZoneByName(ctx, svc, project, zoneName)
+}
+
+func (c *Client) UpdateDNSPrivateZoneLabels(ctx context.Context, baseDomain, project, zoneName string, labels map[string]string) error {
+	params := gcptypes.DNSZoneParams{
+		Project:    project,
+		Name:       zoneName,
+		BaseDomain: baseDomain,
+		IsPublic:   false,
+	}
+	zone, err := c.GetDNSZoneFromParams(ctx, params)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to get DNS Zones")
+		return err
 	}
-	return returnedZone, nil
+	if zone == nil {
+		return fmt.Errorf("failed to find matching DNS zone for %s in project %s", zoneName, project)
+	}
+
+	if zone.Labels == nil {
+		zone.Labels = make(map[string]string)
+	}
+
+	for key, value := range labels {
+		zone.Labels[key] = value
+	}
+
+	if zone.Description != "" {
+		zone.Description = fmt.Sprintf("%s - updated", zone.Description)
+	} else {
+		zone.Description = "updated"
+	}
```

Comment:
Why change the zone description?

Reply:
I wasn't sure what to put there. If the description somehow is  not there it causes this to fail so something must be there. Any ideas of what to change this to in the event that one does not exist ?